### PR TITLE
use readline

### PR DESCRIPTION
**Pull request by mxyng**

### CODE REVIEW

File: cmd/cmd.go [hunk 1]
```diff
@@ -223,17 +229,80 @@ func generate(cmd *cobra.Command, model, prompt string) error {
...
-		if err := generate(cmd, model, scanner.Text()); err != nil {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return err
+	}
+
+	completer := readline.NewPrefixCompleter(
+		readline.PcItem("help"),
+		readline.PcItem("list"),
+		// readline.PcItem("print",
```
Comment (mxyng) [inline cmd/cmd.go:240]: This is a placeholder. Will likely remove before merging

**Review by pdevine — COMMENTED** [review]
I'm super stoked about this, but I think the commands should each start with "/" otherwise it feels like you're going to send the command to the model.

File: cmd/cmd.go [hunk 2]
```diff
@@ -227,17 +231,115 @@ func generate(cmd *cobra.Command, model, prompt string) error {
...
+			args := strings.Fields(line)
+			if len(args) > 1 {
+				switch args[1] {
+				case "history":
+					scanner.HistoryEnable()
+					continue
+				case "nohistory":
+					scanner.HistoryDisable()
+					continue
+				case "verbose":
```
Comment (pdevine) [inline cmd/cmd.go:306]: love this

File: cmd/cmd.go [hunk 3]
```diff
@@ -227,17 +231,115 @@ func generate(cmd *cobra.Command, model, prompt string) error {
...
+							scanner.SetVimMode(true)
+							continue
+						case "emacs", "default":
+							scanner.SetVimMode(false)
+							continue
+						}
+					}
+				}
+			}
+		case line == "/clear":
```
Comment (pdevine) [inline cmd/cmd.go:325]: we can cut this out for now if it's causing problems

**Review by pdevine — APPROVED** [review]

**Pull request merged**